File: data/src/main/java/com/marvelcomics/brito/data/datasource/remote/response/CharacterResponse.java
package com.marvelcomics.brito.data.datasource.remote.response;

import com.google.gson.annotations.SerializedName;

public class CharacterResponse {

    @SerializedName("id")
    private int id;
    @SerializedName("name")
    private String name;
    @SerializedName("description")
    private String description;
    @SerializedName("modified")
    private String modified;
    @SerializedName("thumbnail")
    private ThumbnailResponse thumbnailResponse;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getModified() {
        return modified;
    }

    public void setModified(String modified) {
        this.modified = modified;
    }

    public ThumbnailResponse getThumbnailResponse() {
        return thumbnailResponse;
    }

    public void setThumbnailResponse(ThumbnailResponse thumbnailResponse) {
        this.thumbnailResponse = thumbnailResponse;
    }
}

File: data-remote/src/main/java/com/marvelcomics/brito/data_remote/datasource/response/CharacterResponse.kt
package com.marvelcomics.brito.data_remote.datasource.response

import com.google.gson.annotations.SerializedName
import com.marvelcomics.brito.data_remote.datasource.response.model.RemoteMarvelContainerResponse
import com.marvelcomics.brito.entity.CharacterEntity

class CharacterResponse {
    @SerializedName("id")
    var id = 0
    @SerializedName("name")
    var name: String? = null
    @SerializedName("description")
    var description: String? = null
    @SerializedName("modified")
    var modified: String? = null
    @SerializedName("thumbnail")
    var thumbnailResponse: ThumbnailResponse? = null
}

fun RemoteMarvelContainerResponse<CharacterResponse>.fromResponseToEntity(): List<CharacterEntity>? {
    return this.remoteMarvelDataResponse?.results?.map { characterResponse ->
        CharacterEntity(
            characterResponse.id,
            characterResponse.name,
            characterResponse.description,
            characterResponse.thumbnailResponse?.fromResponseToEntity()
        )
    }
}
